src/universe/visualization/img_tools.py

```diff
@@ -14,3 +18,313 @@ def rotate(img: np.ndarray, angle: float) -> np.ndarray:
     rot_mat = cv.getRotationMatrix2D((cols // 2, rows // 2), angle, 1)
 
     return cv.warpAffine(img, rot_mat, (cols, rows))
+
+class _Layer:
+    """
+    A layer that can be drawn on an image.
+    """
+
+    def __init__(self, color: Tuple[int, int, int], opacity: int, name: str = None):
```

Comment:
In type hinting, if a variable can be None, you should wrap its type with Optional:
```suggestion
    def __init__(self, color: Tuple[int, int, int], opacity: int, name: Optional[str] = None):
```
Or the newer syntax:
```suggestion
    def __init__(self, color: Tuple[int, int, int], opacity: int, name: str | None = None):
```